### PR TITLE
feat: add view Recurring Deposits Account Page

### PR DESCRIPTION
**Pull request by muskankhedia**
## Description
 - Adds view recurring deposit account page

## Related issues and discussion
#227 

## Screenshots, if any
Screenshot from 2020-07-03 10-38-31

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/clients/clients-view/general-tab/general-tab.component.html [hunk 1]
```diff
@@ -555,7 +555,7 @@ <h3>Shares Accounts</h3>
 
     <tr mat-header-row *matHeaderRowDef="openSharesColumns"></tr>
     <tr mat-row *matRowDef="let row; columns: openSharesColumns;"
-      [routerLink]="['../', 'sharesaccounts', row.id, 'transactions']"></tr>
```
Comment (karantakalkar) [inline src/app/clients/clients-view/general-tab/general-tab.component.html:558]: why this?

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+              <td><span
+                  *ngIf="recurringDepositsAccountData.summary.totalDeposits">{{recurringDepositsAccountData.currency.displaySymbol}}&nbsp;{{recurringDepositsAccountData.summary.totalDeposits | number}}</span>
+                <span
+                  *ngIf="!recurringDepositsAccountData.summary.totalDeposits">{{recurringDepositsAccountData.currency.displaySymbol}}&nbsp;0</span>
+              </td>
+            </tr>
+            <tr>
+              <td>Withdrwals till Date</td>
+              <td><span
+                  *ngIf="recurringDepositsAccountData.summary.totalWithdrawals">{{recurringDepositsAccountData.currency.displaySymbol}}&nbsp;{{recurringDepositsAccountData.summary.totalWithdrawals | number}}</span>
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:145]: single ngIf on tr will do

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+  </mat-card-header>
+
+  <mat-card-content class="content">
+
+    <div class="recurringDeposits-account-tables" fxLayout="row" fxLayoutGap="2%">
+
+      <div fxFlex="49%"
+        *ngIf="!recurringDepositsAccountData.status.rejected && !recurringDepositsAccountData.status.submittedAndPendingApproval">
+        <h4 class="table-headers">Savings Details</h4>
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:65]: reccuring deposits details

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  ngOnInit() {
+    this.setConditionalButtons();
+  }
+
+  formatFrequencyType(val: string) {
+    const type = val.split(',')[-1];
+    return type.charAt(0).toUpperCase() + type.slice(1);
+  }
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts:52]: use titlecase pipe, also dont use custom functions just to modify strings in the view

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+              <td>Interest compounding period</td>
+              <td><span>{{recurringDepositsAccountData.interestPostingPeriodType.value}}</span></td>
+            </tr>
+            <tr>
+              <td>Interest calculated using</td>
+              <td><span>{{recurringDepositsAccountData.interestCalculationType.value}}</span></td>
+            </tr>
+            <tr>
+              <td># Days in Year</td>
+              <td><span>{{recurringDepositsAccountData.interestCalculationDaysInYearType.value}}</span></td>
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:185]: Are spans useful here?

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    </mat-card-title-group>
+
+    <mat-card-actions class="account-actions">
+
+      <ng-container *ngFor="let button of buttonConfig.singleButtons">
+        <button mat-raised-button >
+          <i class="{{button.icon}}"></i> {{button.name}}</button>
+      </ng-container>
+  
+      <ng-container *ngIf="buttonConfig.options.length > 0">
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:46]:
```suggestion
      <ng-container *ngIf="buttonConfig.options.length">
```

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+              <td>Activation date</td>
+              <td>{{recurringDepositsAccountData.timeline.activatedOnDate | date}}</td>
+            </tr>
+            <tr>
+              <td>Maturity Date</td>
+              <td><span>{{recurringDepositsAccountData.maturityDate | date}}</span></td>
+            </tr>
+            <tr
+              *ngIf="recurringDepositsAccountData.timeline.closedOnDate || recurringDepositsAccountData.summary.totalWitddrawals">
+              <td *ngIf="recurringDepositsAccountData.timeline.closedOnDate">
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:83]: single ngIf on tr, also this logic in community app seems flawed to me

File: src/app/recurring-deposits/recurring-deposits.module.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import { PipesModule } from '../pipes/pipes.module';
+import { DirectivesModule } from '../directives/directives.module';
+
+/** Custom Components */
+import { RecurringDepositsAccountViewComponent } from './recurring-deposits-account-view/recurring-deposits-account-view.component';
+
+/**
+ * RecurringDeposits Module
+ *
+ * All components related to RecurringDeposits functions should be declared here.
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits.module.ts:17]:
```suggestion
 * All components related to Recurring Deposits functions should be declared here.
```

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+   */
+  setConditionalButtons() {
+    const status = this.recurringDepositsAccountData.status.value;
+    this.buttonConfig = new RecurringDepositsButtonsConfiguration(status);
+    if (this.recurringDepositsAccountData.clientId && this.recurringDepositsAccountData.status.value === 'Matured') {
+      this.buttonConfig.addOption({
+        name: 'Transfer Funds',
+      });
+    }
+
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts:65]:
```suggestion
```

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  recurringDepositsAccountData: any;
+  buttonConfig: RecurringDepositsButtonsConfiguration;
+  charges: any;
+
+  /**
+   * Fetches recurringDeposits account data from `resolve`
+   * @param {ActivatedRoute} route Activated Route
+   * @param {Router} router Router
+   * @param {RecurringDepositsService} recurringDepositsService RecurringDeposits Service
+   */
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts:33]: update comments

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.scss [hunk 11]
```diff
@@ -0,0 +1,53 @@
+.recurringDeposits-account-card {
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.scss:1]: no camelcasing in css, use kebab casing

File: src/app/clients/clients-view/general-tab/general-tab.component.html (hunk 1 above)
Comment (muskankhedia) [reply src/app/clients/clients-view/general-tab/general-tab.component.html:558]: won't you navigate for the closed recurring deposits account?

Comment (muskankhedia) [reply src/app/clients/clients-view/general-tab/general-tab.component.html:558]: won't you navigate for the closed recurring deposits account?

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html (hunk 2 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:145]: I know.. that's a better way of implementation.

File: src/app/clients/clients-view/general-tab/general-tab.component.html (hunk 1 above)
Comment (karantakalkar) [reply src/app/clients/clients-view/general-tab/general-tab.component.html:558]: Its the shares account table

**Comment by muskankhedia** [comment]
> You can include rest sih, transactions, charges, incentives as its already been done for savings and fd

As discussed with the mentors, we should make a PR for every feature we are adding and also have a corresponding issue to it. I would also recommend you to follow the same.

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html (hunk 3 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:65]: Please have a look at the community app code.

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html (hunk 6 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:46]: That doesn't work. Please modify in the savings section as well.

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.scss (hunk 11 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.scss:1]: that's for a purpose

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts (hunk 10 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts:33]: will update comments in refactoring

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+            <tr>
+              <td>Period</td>
+              <td>
+                <span>{{recurringDepositsAccountData.depositPeriod}}&nbsp;{{recurringDepositsAccountData.depositPeriodFrequency.value}}</span>
+              </td>
+            </tr>
+            <tr>
+              <td>Deposit Frequency</td>
+              <td><span>{{recurringDepositsAccountData.recurringFrequency}}
+                  &nbsp;{{recurringDepositsAccountData.recurringFrequencyType.code}}
```
Comment (karantakalkar) [inline src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:96]: see your screenshot for this

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html (hunk 7 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:83]: Yaa.. but there could be some other params as well that could be added later. So unless we don't know the business logic better to go with the community app.

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html (hunk 5 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:185]: Implemented to maintain the consistency

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts (hunk 4 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.ts:52]: presently this function isn't used, need some further work on it.

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.scss (hunk 11 above)
Comment (karantakalkar) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.scss:1]: ?

File: src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html (hunk 12 above)
Comment (muskankhedia) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:96]: As the web-app doesn't support translation for now. I have to choose the path of implementing through functions. And that's the reason that function has been written but presently that is giving some errors. Will modify it in the upcoming PR.

Comment (karantakalkar) [reply src/app/recurring-deposits/recurring-deposits-account-view/recurring-deposits-account-view.component.html:96]: use .value instead of .code

**Pull request merged**